File: artifacts/tulipindicators/decompose/src/log10.c
int ti_log10_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_log10(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *in1 = inputs[0];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = log10(in1[i]);
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/manual_translation/src/log10.rs
use crate::*;

pub fn ti_log10_start(options: Option<&[f64]>) -> i32 {
	println!("\n* ti_log10_start");
    let _ = options; // Explicitly ignore the parameter to match C behavior
    0
}
pub fn ti_log10(
    size: i32,
    inputs: Option<&[&[f64]]>,
    options: Option<&[f64]>,
    outputs: Option<&mut [&mut [f64]]>,
) -> i32 {
	println!("\n* ti_log10");
    let inputs = inputs.unwrap();
    let outputs = outputs.unwrap();
    let in1 = inputs[0];
    let output = &mut outputs[0];
    
    for i in 0..size as usize {
        output[i] = in1[i].log10();
    }
    
    0
}
